cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```diff
@@ -71,6 +81,184 @@ public Schema getSchema() {
+    LogicalType logicalType = validateAndGetLogicalType(fieldName, EnumSet.of(LogicalType.TIME_MILLIS,
+                                                                              LogicalType.TIME_MICROS));
+    Object value = fields.get(fieldName);
+    if (value == null) {
+      return null;
+    }
+
+    if (logicalType == LogicalType.TIME_MILLIS) {
+      return LocalTime.ofNanoOfDay(TimeUnit.MILLISECONDS.toNanos(((Integer) value)));
+    }
+
+    return LocalTime.ofNanoOfDay(TimeUnit.MICROSECONDS.toNanos((Long) value));
+  }
+
+  /**
+   * Get the {@link ZonedDateTime} with timezone 'UTC' for the specified field.
+   * The field must have {@link Schema.LogicalType#TIMESTAMP_MILLIS} or {@link Schema.LogicalType#TIMESTAMP_MICROS}
+   * as its logical type.
+   *
+   * @param fieldName zoned date time field to get.
+   * @return value of the field as a {@link ZonedDateTime}
+   * logical type timestamp
+   * @throws UnexpectedFormatException if the field name is not found in the map of fields or the provided field
+   *                                   is not of {@link LogicalType#TIMESTAMP_MILLIS} or
+   *                                   {@link LogicalType#TIMESTAMP_MICROS} type.
+   */
+  @Nullable
+  public ZonedDateTime getTimestamp(String fieldName) {
+    return getTimestamp(fieldName, ZoneId.ofOffset("UTC", ZoneOffset.UTC));
+  }
+
+  /**
+   * Get the {@link ZonedDateTime} with provided timezone.
+   * The field must have {@link Schema.LogicalType#TIMESTAMP_MILLIS} or {@link Schema.LogicalType#TIMESTAMP_MICROS}
+   * as its logical type.
+   *
+   * @param fieldName zoned date time field to get.
+   * @param zoneId zone id for the field
+   * @return value of the field as a {@link ZonedDateTime}
+   * @throws UnexpectedFormatException if the field name is not found in the map of fields or the provided field
+   *                                   is not of {@link LogicalType#TIMESTAMP_MILLIS} or
+   *                                   {@link LogicalType#TIMESTAMP_MICROS} type.
+   */
+  @Nullable
+  public ZonedDateTime getTimestamp(String fieldName, ZoneId zoneId) {
+    LogicalType logicalType = validateAndGetLogicalType(fieldName, EnumSet.of(LogicalType.TIMESTAMP_MILLIS,
+                                                                              LogicalType.TIMESTAMP_MICROS));
+    Object value = fields.get(fieldName);
+    if (value == null) {
+      return null;
+    }
+
+    if (logicalType == LogicalType.TIMESTAMP_MILLIS) {
+      return getZonedDateTime((long) value, zoneId, TimeUnit.MILLISECONDS);
+    }
+
+    return getZonedDateTime((long) value, zoneId, TimeUnit.MICROSECONDS);
+  }
+
+  /**
+   * Get zoned date and time represented by the field.
+   *
+   * @param ts timestamp from which
+   * @param zoneId zone id for the field
+   * @param unit time unit for ts
+   * @return {@link ZonedDateTime} represented by field.
+   */
+  private ZonedDateTime getZonedDateTime(long ts, ZoneId zoneId, TimeUnit unit) {
```

Comment:
Have `unit` comes before `zoneId` to be consistent with all Java API that takes a time and a time unit.